### PR TITLE
Additional MariaDB banners 

### PR DESCRIPTION
**Pull request by jpincombe-r7**
## Description
Additional MariaDB banners that current recog is not matching against

## How Has This Been Tested?
Running the unit tests with 
bin/recog_verify xml/mysql_banners.xml

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] I have updated the documentation accordingly (or changes are not required).
- [ ] I have added tests to cover my changes (or new tests are not required).
- [ ] All new and existing tests passed.

### CODE REVIEW

File: xml/mysql_banners.xml [hunk 1]
```diff
@@ -1171,6 +1361,42 @@
     <param pos="0" name="service.family" value="MySQL"/>
     <param pos="0" name="service.product" value="MariaDB"/>
   </fingerprint>
+  <fingerprint pattern="^(?:\d{1,2}\.\d{1,3}\.[a-f\d]{1,3}-)(\d*\.\d*\.\d*)-MariaDB.*-*~bionic?(?:-log)?$" flags="REG_ICASE">
+	<example service.version="10.2.17">5.5.5-10.2.17-MariaDB-1:10.2.17+maria~bionic</example>
```
Comment (rkirk-nos) [inline xml/mysql_banners.xml:1469]: missing description

File: xml/mysql_banners.xml [hunk 2]
```diff
@@ -1171,6 +1361,42 @@
...
+    <param pos="0" name="service.family" value="MySQL"/>
+    <param pos="0" name="service.product" value="MariaDB"/>
+    <param pos="0" name="os.vendor" value="Debian"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux"/>
+    <param pos="0" name="os.cpe23" value="cpe:/o:debian:debian_linux:9.0"/>
+  </fingerprint>
+  <fingerprint pattern="^(?:\d{1,2}\.\d{1,3}\.[a-f\d]{1,3}-)(\d*\.\d*\.\d*)-MariaDB\S*deb\S*$" flags="REG_ICASE">
+   <example service.version="10.1.23">5.5.5-10.1.23-MariaDB-9+deb9u1</example>
+   <description>MariaDB runnin on Debian 9.1(Stretch)</description>
```
Comment (rkirk-nos) [inline xml/mysql_banners.xml:1494]: description should come first for consistency

File: xml/mysql_banners.xml [hunk 3]
```diff
@@ -903,6 +904,130 @@
...
+    <param pos="0" name="service.family" value="MySQL"/>
+    <param pos="0" name="service.product" value="MariaDB"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:mariadb:mariadb:{service.version}"/>
+    <param pos="0" name="os.vendor" value="Debian"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux"/>
+    <param pos="0" name="os.version" value="10.03"/>
+    <param pos="0" name="os.cpe23" value="cpe:/o:canonical:debian_linux:10.03"/>
+  </fingerprint>
+    <fingerprint pattern="^(?:\d{1,2}\.\d{1,3}\.[a-f\d]{1,3}-)?(\d{1,2}\.\d{1,3}\.[a-f\d]{1,4})-MariaDB.+~yakkety(?:-log)?$" flags="REG_ICASE">
```
Comment (pfitzsimons-r7) [inline xml/mysql_banners.xml]: formatting is off 👍

File: xml/mysql_banners.xml [hunk 4]
```diff
@@ -903,6 +904,130 @@
...
+    <description>MariaDB</description>
+    <example service.version="10.0.24">5.5.5-10.0.24-MariaDB-1~vivid</example>
+    <example service.version="10.1.10">5.5.5-10.1.10-MariaDB-1~vivid-log</example>
+    <param pos="1" name="service.version"/>
+    <param pos="0" name="service.vendor" value="MariaDB"/>
+    <param pos="0" name="service.family" value="MySQL"/>
+    <param pos="0" name="service.product" value="MariaDB"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:mariadb:mariadb:{service.version}"/>
+  </fingerprint>
+      <fingerprint pattern="^(?:\d{1,2}\.\d{1,3}\.[a-f\d]{1,3}-)?(\d{1,2}\.\d{1,3}\.[a-f\d]{1,4})-MariaDB.+~vivid-wsrep(?:-log)?$" flags="REG_ICASE">
```
Comment (pfitzsimons-r7) [inline xml/mysql_banners.xml]: indentation

File: xml/mysql_banners.xml [hunk 5]
```diff
@@ -1065,6 +1282,38 @@
...
+    <param pos="0" name="service.product" value="MariaDB"/>
+    <param pos="0" name="service.edition" value="Galera Cluster"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:mariadb:mariadb:{service.version}"/>
+    <param pos="0" name="os.vendor" value="Ubuntu"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux"/>
+    <param pos="0" name="os.version" value="17.04"/>
+    <param pos="0" name="os.cpe23" value="cpe:/o:debian:ubuntu_linux:17.04"/>
+  </fingerprint>
+    <fingerprint pattern="^(?:\d{1,2}\.\d{1,3}\.[a-f\d]{1,3}-)?(\d{1,2}\.\d{1,3}\.[a-f\d]{1,4})-MariaDB.+~artful(?:-log)?$" flags="REG_ICASE">
```
Comment (pfitzsimons-r7) [inline xml/mysql_banners.xml]: indentation

File: xml/mysql_banners.xml [hunk 6]
```diff
@@ -1035,6 +1192,66 @@
...
+    <param pos="0" name="service.product" value="MariaDB"/>
+    <param pos="0" name="service.edition" value="Galera Cluster"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:mariadb:mariadb:{service.version}"/>
+    <param pos="0" name="os.vendor" value="Ubuntu"/>
+    <param pos="0" name="os.family" value="Linux"/>
+    <param pos="0" name="os.product" value="Linux"/>
+    <param pos="0" name="os.version" value="10.10	"/>
+    <param pos="0" name="os.cpe23" value="cpe:/o:canonical:ubuntu_linux:10.04"/>
+  </fingerprint>
+    <fingerprint pattern="^(?:\d{1,2}\.\d{1,3}\.[a-f\d]{1,3}-)?(\d{1,2}\.\d{1,3}\.[a-f\d]{1,4}\-\d{1,2})-MariaDB-enterprise(?:-log)?$" flags="REG_ICASE">
```
Comment (rcullen-r7) [inline xml/mysql_banners.xml]: alignment is off

Comment (rcullen-r7) [reply xml/mysql_banners.xml:1240]: paddy beat me to it

**Comment by jpincombe-r7** [comment]
Going to be closing this pr as i have opened a new one at the rapid7 fork
https://github.com/rapid7/recog/pull/264

**Pull request closed without merging**